### ISSUE TITLE
microdom assigns incorrect parent values to top-level nodes in malformed HTML strings

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2298]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)| @jml reported|
|-|-|
|Trac ID|trac#2298|
|Type|enhancement|
|Created|2006-12-15 04:59:22Z|

For example:
```
>>> from twisted.web.microdom import parseString
>>> html = "<br/><br/>"
>>> d = parseString(html, beExtremelyLenient=1)
>>> d.documentElement
Element('html')
>>> d.documentElement.firstChild()
Element('br', filename='<xmlfile />', markpos=(1, 5))
>>> print d.documentElement.firstChild().parentNode
None
```

The last value should be `d.documentElement`, not None.

<details><summary>Searchable metadata</summary>

```
trac-id__2298 2298
type__enhancement enhancement
reporter__jml jml
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__web web
keywords__ 
time__1166158762000000 1166158762000000
changetime__1166244366000000 1166244366000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)|@jml set owner to @exarkun|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun set owner to @jml|
|-|-|

Nice.  Fix looks good, test coverage is great.  Merge away.

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_3"></a>|@jml set status to `closed`|
|-|-|

(In [18911]) Ensure root nodes in malformed HTML documents have correct parentNodes
 * Fixes [#2298](https://github.com/twisted/twisted/issues/2298)
 * Author: jml
 * Reviewer: exarkun

Previously, if an HTML document had multiple root nodes, microdom would 
give those root nodes a parent of None, even though it listed them as 
children of an <html> element. This patch sets the parent correctly.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
